utopia-remix/app/components/projectActionContextMenu.tsx

```diff
@@ -16,16 +24,20 @@ type ContextMenuEntry =
   | 'separator'
 
 export const ProjectContextMenu = React.memo(({ project }: { project: ProjectWithoutContent }) => {
-  const deleteFetcher = useFetcherWithOperation(operation(project, 'delete'))
-  const destroyFetcher = useFetcherWithOperation(operation(project, 'destroy'))
-  const restoreFetcher = useFetcherWithOperation(operation(project, 'restore'))
-  const renameFetcher = useFetcherWithOperation(operation(project, 'rename'))
+  const deleteFetcher = useFetcherWithOperation(project.proj_id, 'delete')
+  const destroyFetcher = useFetcherWithOperation(project.proj_id, 'destroy')
+  const restoreFetcher = useFetcherWithOperation(project.proj_id, 'restore')
+  const renameFetcher = useFetcherWithOperation(project.proj_id, 'rename')
 
   const selectedCategory = useProjectsStore((store) => store.selectedCategory)
 
   const deleteProject = React.useCallback(
     (projectId: string) => {
-      deleteFetcher.submit({}, { method: 'POST', action: `/internal/projects/${projectId}/delete` })
+      deleteFetcher.submit(
+        operationDelete(project),
```

Comment:
Since we're creating a fetcher specifically for each type of operation, it's probably worth trying to create those in a way that means we don't have to supply the operation itself (i.e. we know this is a deletion only fetcher, so we should be able to supply it with the type of operation, method, and action at the time of creating it, and just tell it which project to delete when using it)

Reply:
Actually, saying that, I've just realised that we're only supplying the operation type to use in the string `key`, so maybe just ignore the above and leave it as is :)

Reply:
yeah I was about to reply: I had to play around with this a bit last night - the main problem being that the operation should also contain the information it's receiving _when_ submitted, which is unfortunately no available before that (and can obviously change every time :( )